### PR TITLE
Invalidate GQL cache for draft projects on creation

### LINKED ISSUE #937: new draft projects don't show up on my projects list without a refresh
when making a project request, those draft projects show up in the My Projects list, however the gql cache means they won't show up automatically and you need to refresh the page, we should invalidate that cache when creating a new project draft request.

### PR DESCRIPTION
**Pull request by rmunn**
Fixes #937.

This PR is on top of #976 to avoid merge conflicts.

### CODE REVIEW

File: frontend/src/lib/gql/gql-client.ts [hunk 1]
```diff
@@ -61,6 +61,19 @@ function createGqlClient(_gqlEndpoint?: string): Client {
               if (args.input.orgId) {
                 cache.invalidate({__typename: 'OrgById', id: args.input.orgId}, 'projects');
               }
+              if (args.input.id) {
+                cache.invalidate({__typename: 'Project', id: args.input.id});
+                cache.invalidate({__typename: 'DraftProject', id: args.input.id});
+                // Note singular MyProject name for the myProjects query cache, ditto for draft
+                cache.invalidate({__typename: 'MyProject', id: args.input.id});
+                cache.invalidate({__typename: 'MyDraftProject', id: args.input.id});
```
Comment (hahn-kev) [inline frontend/src/lib/gql/gql-client.ts:69]: there's not actually a `MyProject` or `MyDraftProject` type  so those references are unneeded. Take a look at the schema.graphql file to see the types the exist. The `myProjects` query just returns a list of `Projects`

File: frontend/src/lib/gql/gql-client.ts [hunk 2]
```diff
@@ -61,6 +61,19 @@ function createGqlClient(_gqlEndpoint?: string): Client {
...
+              }
+              if (result?.createProject?.createProjectResponse?.id) {
+                cache.invalidate({__typename: 'Project', id: result?.createProject?.createProjectResponse?.id});
+                cache.invalidate({__typename: 'DraftProject', id: result?.createProject?.createProjectResponse?.id});
+                cache.invalidate({__typename: 'MyProject', id: result?.createProject?.createProjectResponse?.id});
+                cache.invalidate({__typename: 'MyDraftProject', id: result?.createProject?.createProjectResponse?.id});
```
Comment (hahn-kev) [inline frontend/src/lib/gql/gql-client.ts:75]: I think only `DraftProject` is needed here.

**Review by hahn-kev — CHANGES_REQUESTED** [review]
nice work, this cache invalidation is a bit of a pain.

File: frontend/src/lib/gql/gql-client.ts (hunk 1 above)
Comment (rmunn) [reply frontend/src/lib/gql/gql-client.ts:69]: And yet, when I comment out the `MyDraftProject` invalidation on line 69, draft projects don't show up in the "My Projects" immediately after clicking on the button. (They do show up on refresh). If I uncomment line 69, then the cache invalidation works as expected.

There's something going on with the typenames in the cache that isn't immediately obvious from looking at `schema.graphql`.

File: frontend/src/lib/gql/gql-client.ts (hunk 2 above)
Comment (rmunn) [reply frontend/src/lib/gql/gql-client.ts:74]: Removed `Project` cache invalidation in commit 27cefe5a. Keeping MyProject and MyDraftProject as mentioned in https://github.com/sillsdev/languageforge-lexbox/pull/977#discussion_r1687802315.

File: frontend/src/lib/gql/gql-client.ts [hunk 3]
```diff
@@ -61,6 +61,19 @@ function createGqlClient(_gqlEndpoint?: string): Client {
               if (args.input.orgId) {
                 cache.invalidate({__typename: 'OrgById', id: args.input.orgId}, 'projects');
               }
+              if (args.input.id) {
+                cache.invalidate({__typename: 'Project', id: args.input.id});
```
[the remark this answers is not in the document]
Comment (rmunn) [reply frontend/src/lib/gql/gql-client.ts]: Removed `Project` cache invalidation in commit 27cefe5a.

**Comment by rmunn** [comment]
@hahn-kev - As we discussed before you left today, I just rewrote your commit ac2c0ec to remove the import alphabetizaton and whitespace changes, so that code review will be easier as the PR has fewer changed lines. (Result is commit a1a5390). The two commits [should be otherwise identical](https://github.com/sillsdev/languageforge-lexbox/compare/ac2c0ecfdefbcd0d3492c10faea5587fdc1088df..a1a5390a4c2eb3280c23ee91293299af12211fa8).

**Comment by rmunn** [comment]
Continued testing shows that both draft projects and normal projects show up in the user's home page... but **NOT** the admin dashboard. More work is needed.

**Comment by rmunn** [comment]
@hahn-kev - Commit e5cfdd72 now gets new projects showing up in the admin dashboard. This should be ready to test and review again.

**Review by hahn-kev — APPROVED** [review]

**Pull request merged**